From 1ea1dd2095e63cef34edbebb729edc687f410a96 Mon Sep 17 00:00:00 2001
From: Alexander Barton <alex@barton.de>
Date: Mon, 16 May 2011 18:26:56 +0200
Subject: [PATCH 2/2] ngircd: whitespace fixes

---
 modules/protocol/ngircd.cpp |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/modules/protocol/ngircd.cpp b/modules/protocol/ngircd.cpp
index 6e1f21f..e546d05 100644
--- a/modules/protocol/ngircd.cpp
+++ b/modules/protocol/ngircd.cpp
@@ -266,11 +266,11 @@ bool event_chaninfo(const Anope::string &source, const std::vector<Anope::string
 		c = new Channel(params[0]);
 
 	Anope::string modes = params[1];
-	
+
 	if (params.size() == 3)
 	{
 		c->ChangeTopicInternal(source, params[2], Anope::CurTime);
-	} 
+	}
 	else if (params.size() == 5)
 	{
 		for (size_t i = 0, end = params[1].length(); i < end; ++i)
@@ -307,7 +307,7 @@ bool event_njoin(const Anope::string &source, const std::vector<Anope::string> &
 		c = new Channel(params[0], Anope::CurTime);
 		c->SetFlag(CH_SYNCING);
 	}
-	
+
 	while (sep.GetToken(buf))
 	{
 		std::list<ChannelMode *> Status;
@@ -415,9 +415,9 @@ bool event_376(const Anope::string &source, const std::vector<Anope::string> &pa
 
 class ProtongIRCd : public Module
 {
-	Message message_kick, message_pass, message_njoin, message_chaninfo, message_005, 
+	Message message_kick, message_pass, message_njoin, message_chaninfo, message_005,
 		message_442, message_376;
-	
+
 	ngIRCdProto ircd_proto;
 	ngIRCdIRCdMessage ircd_message;
 
@@ -461,7 +461,7 @@ class ProtongIRCd : public Module
 	{
 		this->SetAuthor("Anope");
 		this->SetType(PROTOCOL);
-		
+
 		Capab.SetFlag(CAPAB_QS);
 
 		pmodule_ircd_var(myIrcd);
-- 
1.7.2.5

