From 1914a36b83b1fc6b4678ef261a1a06eefab9a0ca Mon Sep 17 00:00:00 2001
From: Alexander Barton <alex@barton.de>
Date: Fri, 26 Aug 2011 17:51:37 +0200
Subject: [PATCH 06/16] ngircd: support user mode "R" and channel mode "R"

---
 modules/protocol/ngircd.cpp |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/modules/protocol/ngircd.cpp b/modules/protocol/ngircd.cpp
index 3e5beb3..7f4186e 100644
--- a/modules/protocol/ngircd.cpp
+++ b/modules/protocol/ngircd.cpp
@@ -449,26 +449,27 @@ class ProtongIRCd : public Module
 		ModeManager::AddUserMode(new UserMode(UMODE_INVIS, 'i'));
 		ModeManager::AddUserMode(new UserMode(UMODE_OPER, 'o'));
 		ModeManager::AddUserMode(new UserMode(UMODE_RESTRICTED, 'r'));
+		ModeManager::AddUserMode(new UserMode(UMODE_REGISTERED, 'R'));
 		ModeManager::AddUserMode(new UserMode(UMODE_SNOMASK, 's'));
 		ModeManager::AddUserMode(new UserMode(UMODE_WALLOPS, 'w'));
 		ModeManager::AddUserMode(new UserMode(UMODE_CLOAK, 'x'));
 
-		/* b/e/I */
+		/* Add modes for ban and invite lists */
 		ModeManager::AddChannelMode(new ChannelModeList(CMODE_BAN, 'b'));
 		ModeManager::AddChannelMode(new ChannelModeList(CMODE_INVITEOVERRIDE, 'I'));
 
-		/* v/h/o/a/q */
+		/* Add channel user modes */
 		ModeManager::AddChannelMode(new ChannelModeStatus(CMODE_VOICE, 'v', '+'));
 		ModeManager::AddChannelMode(new ChannelModeStatus(CMODE_OP, 'o', '@'));
 
 		/* Add channel modes */
-		// channel modes: biIklmnoPstvz
 		ModeManager::AddChannelMode(new ChannelMode(CMODE_INVITE, 'i'));
 		ModeManager::AddChannelMode(new ChannelModeKey('k'));
 		ModeManager::AddChannelMode(new ChannelModeParam(CMODE_LIMIT, 'l'));
 		ModeManager::AddChannelMode(new ChannelMode(CMODE_MODERATED, 'm'));
 		ModeManager::AddChannelMode(new ChannelMode(CMODE_NOEXTERNAL, 'n'));
 		ModeManager::AddChannelMode(new ChannelMode(CMODE_PERM, 'P'));
+		ModeManager::AddChannelMode(new ChannelMode(CMODE_REGISTEREDONLY, 'R'));
 		ModeManager::AddChannelMode(new ChannelMode(CMODE_SECRET, 's'));
 		ModeManager::AddChannelMode(new ChannelMode(CMODE_TOPIC, 't'));
 		ModeManager::AddChannelMode(new ChannelMode(CMODE_SSL, 'z'));
-- 
1.7.8.3

