From 99c18cafdee28bfb17fad5f0526b3ed5d1f5f312 Mon Sep 17 00:00:00 2001
From: Alexander Barton <alex@barton.de>
Date: Sat, 31 Dec 2011 16:17:50 +0100
Subject: [PATCH 12/16] ngircd: let Anope know that channel mode "r" is
 supported

---
 modules/protocol/ngircd.cpp |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/modules/protocol/ngircd.cpp b/modules/protocol/ngircd.cpp
index 9c26ec8..6155667 100644
--- a/modules/protocol/ngircd.cpp
+++ b/modules/protocol/ngircd.cpp
@@ -512,6 +512,7 @@ class ProtongIRCd : public Module
 		ModeManager::AddChannelMode(new ChannelMode(CMODE_MODERATED, 'm'));
 		ModeManager::AddChannelMode(new ChannelMode(CMODE_NOEXTERNAL, 'n'));
 		ModeManager::AddChannelMode(new ChannelMode(CMODE_PERM, 'P'));
+		ModeManager::AddChannelMode(new ChannelModeRegistered('r'));
 		ModeManager::AddChannelMode(new ChannelMode(CMODE_REGISTEREDONLY, 'R'));
 		ModeManager::AddChannelMode(new ChannelMode(CMODE_SECRET, 's'));
 		ModeManager::AddChannelMode(new ChannelMode(CMODE_TOPIC, 't'));
-- 
1.7.8.3

