From 5a19b69f0daceb5b12ec751bc919519a7f712f2d Mon Sep 17 00:00:00 2001
From: Alexander Barton <alex@barton.de>
Date: Sun, 15 Jan 2012 13:36:14 +0100
Subject: [PATCH 13/16] ngircd: Update copyright notice

---
 modules/protocol/ngircd.cpp |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/modules/protocol/ngircd.cpp b/modules/protocol/ngircd.cpp
index 6155667..024c61d 100644
--- a/modules/protocol/ngircd.cpp
+++ b/modules/protocol/ngircd.cpp
@@ -1,7 +1,8 @@
-/* ngIRCd IRCD functions
+/*
+ * ngIRCd Protocol module for Anope IRC Services
  *
- * (C) 2003-2011 Anope Team
- * Contact us at team@anope.org
+ * (C) 2011-2012 Alexander Barton <alex@barton.de>
+ * (C) 2011 Anope Team <team@anope.org>
  *
  * Please read COPYING and README for further details.
  *
-- 
1.7.8.3

