From 3a61b190db79848d4519296432ebb2ab714c42b7 Mon Sep 17 00:00:00 2001
From: Alexander Barton <alex@barton.de>
Date: Sun, 22 Jan 2012 19:06:34 +0100
Subject: [PATCH 15/16] ngircd: ngIRCd supports channel mode 'e' now

---
 modules/protocol/ngircd.cpp |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/modules/protocol/ngircd.cpp b/modules/protocol/ngircd.cpp
index 3bc3812..0f87cbd 100644
--- a/modules/protocol/ngircd.cpp
+++ b/modules/protocol/ngircd.cpp
@@ -504,8 +504,9 @@ class ProtongIRCd : public Module
 		ModeManager::AddUserMode(new UserMode(UMODE_WALLOPS, 'w'));
 		ModeManager::AddUserMode(new UserMode(UMODE_CLOAK, 'x'));
 
-		/* Add modes for ban and invite lists */
+		/* Add modes for ban, exception, and invite lists */
 		ModeManager::AddChannelMode(new ChannelModeList(CMODE_BAN, 'b'));
+		ModeManager::AddChannelMode(new ChannelModeList(CMODE_EXCEPT, 'e'));
 		ModeManager::AddChannelMode(new ChannelModeList(CMODE_INVITEOVERRIDE, 'I'));
 
 		/* Add channel user modes */
-- 
1.7.8.3

